packages/core/core/src/RequestTracker.js

```diff
@@ -1543,55 +1548,94 @@ async function loadRequestGraph(options): Async<RequestGraph> {
+        });
+      }, 5000);
+      let startTime = Date.now();
+      let events = await options.inputFS.getEventsSince(
+        options.watchDir,
+        snapshotPath,
+        opts,
+      );
+      clearTimeout(timeout);
+
+      logger.verbose({
         origin: '@parcel/core',
-        message: `Retrieving file system events since last build...\nThis can take upto a minute after branch changes or npm/yarn installs.`,
+        message: `File system event count: ${events.length}`,
+        meta: {
+          trackableEvent: 'watcher_events_count',
+          watcherEventCount: events.length,
+          duration: Date.now() - startTime,
+        },
       });
-    }, 5000);
-    let startTime = Date.now();
-    let events = await options.inputFS.getEventsSince(
-      options.watchDir,
-      snapshotPath,
-      opts,
-    );
-    clearTimeout(timeout);
 
-    logger.verbose({
-      origin: '@parcel/core',
-      message: `File system event count: ${events.length}`,
-      meta: {
-        trackableEvent: 'watcher_events_count',
-        watcherEventCount: events.length,
-        duration: Date.now() - startTime,
-      },
-    });
+      requestGraph.invalidateUnpredictableNodes();
+      requestGraph.invalidateOnBuildNodes();
+      requestGraph.invalidateEnvNodes(options.env);
+      requestGraph.invalidateOptionNodes(options);
 
-    requestGraph.invalidateUnpredictableNodes();
-    requestGraph.invalidateOnBuildNodes();
-    requestGraph.invalidateEnvNodes(options.env);
-    requestGraph.invalidateOptionNodes(options);
-
-    try {
       await requestGraph.respondToFSEvents(
         options.unstableFileInvalidations || events,
         options,
         10000,
       );
       return requestGraph;
     } catch (e) {
+      if (timeout != null) {
+        clearTimeout(timeout);
+      }
+      let additionalMessage;
+      let logEvent;
+      if (e.message && e.message.includes('invalid clockspec')) {
+        // Note this assumes only one snapshot file per cache and will potentially break
+        // if multiple snapshot files exist in the cache directory
+        const snapshotFile = readdirSync(options.cacheDir).find(file =>
+          file.endsWith('.txt'),
+        );
+        logEvent = {
+          origin: '@parcel/core',
+          message: `Error reading clockspec from snapshot, building with clean cache.`,
+          meta: {
+            additionalMessage: ``,
+            trackableEvent: 'invalid_clockspec_error',
+          },
+        };
+        if (snapshotFile != null) {
+          additionalMessage = readFileSync(
+            path.join(options.cacheDir, snapshotFile),
+          ).toString('utf-8');
+          logEvent.meta.additionalMessage = `Watchman failed reading clockspec, snapshot: ${additionalMessage}`;
+        }
+      }
+      if (!(e instanceof FSBailoutError)) {
```

Comment:
I suppose we also want to log that we bailed out?

Reply:
That happens elsewhere, hence the FSBailoutError addition